### PR TITLE
lunar-client: adapt and update to 3.0.7

### PR DESCRIPTION
**Pull request by surfaceflinger**
###### Description of changes

Bumped the version to 3.0.7.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by surfaceflinger** [comment]
this update changed icon name from lunarclient.png to a very generic launcher.png, not sure if this will conflict with anything but I left it as it is

**Review by LostAttractor — APPROVED** [review]
The icon name change didn't cause any problem on my system, it's not clear if there is something else

File: pkgs/games/lunar-client/default.nix [hunk 1]
```diff
@@ -2,12 +2,12 @@
...
 let
   name = "lunar-client";
-  version = "2.15.1";
+  version = "2.16.1";
 
   desktopItem = makeDesktopItem {
     name = "lunar-client";
     exec = "lunar-client";
-    icon = "lunarclient";
+    icon = "launcher";
```
Comment (SuperSandro2000) [inline pkgs/games/lunar-client/default.nix:10]: Isn't that name super generic and could easily clash?

Comment (surfaceflinger) [reply pkgs/games/lunar-client/default.nix]: Should be better now :)

File: pkgs/games/lunar-client/default.nix [hunk 2]
```diff
@@ -30,7 +30,10 @@ appimageTools.wrapType1 rec {
   extraInstallCommands = ''
     mkdir -p $out/share/applications
     cp ${desktopItem}/share/applications/* $out/share/applications
```
[the remark this answers is not in the document]
Comment (surfaceflinger) [reply pkgs/games/lunar-client/default.nix:32]: How could this be done with appimageTools wrap?

**Comment by surfaceflinger** [comment]
Lunar Client 3.0.5 got released and basically changed many things. I adapted and updated the derivation, also added myself as a maintainer. Not sure if this can be in a single commit like this.

File: pkgs/games/lunar-client/default.nix [hunk 3]
```diff
@@ -1,45 +1,37 @@
...
-    genericName = "Minecraft Client";
-    categories = [ "Game" ];
-  };
-
-  appimageContents = appimageTools.extract {
-    inherit name src;
-  };
+  pname = "lunar-client";
+  version = "3.0.6";
+  name = "${pname}-${version}";
```
Comment (SuperSandro2000) [inline pkgs/games/lunar-client/default.nix:9]: We should drop name as it isn't really required anymore

File: pkgs/games/lunar-client/default.nix [hunk 4]
```diff
@@ -1,45 +1,37 @@
...
 
   src = fetchurl {
     url = "https://launcherupdates.lunarclientcdn.com/Lunar%20Client-${version}.AppImage";
-    name = "lunar-client.AppImage";
-    hash = "sha256-8F6inLctNLCrTvO/f4IWHclpm/6vqW44NKbct0Epp4s=";
+    hash = "sha256-ujDiGfs2py/Sanh1sgXGLfP4ergm3O4p90B4M3WgU54=";
   };
+
+  appimageContents = appimageTools.extract { inherit name src; };
```
Comment (SuperSandro2000) [inline pkgs/games/lunar-client/default.nix:16]:
```suggestion
  appimageContents = appimageTools.extract { inherit pname version src; };
```

File: pkgs/games/lunar-client/default.nix [hunk 5]
```diff
@@ -1,45 +1,37 @@
...
-appimageTools.wrapType1 rec {
-  inherit name src;
+appimageTools.wrapType2 rec {
+  inherit pname version src;
 
   extraInstallCommands = ''
-    mkdir -p $out/share/applications
-    cp ${desktopItem}/share/applications/* $out/share/applications
-    cp -r ${appimageContents}/usr/share/icons/ $out/share/
+    mv $out/bin/{${name},${pname}}
```
Comment (SuperSandro2000) [inline pkgs/games/lunar-client/default.nix:22]:
```suggestion
    mv $out/bin/{${pname}-${version},${pname}}
```

File: pkgs/games/lunar-client/default.nix [hunk 6]
```diff
@@ -1,45 +1,37 @@
...
+    install -Dm444 ${appimageContents}/launcher.desktop $out/share/applications/${pname}.desktop
+    install -Dm444 ${appimageContents}/launcher.png $out/share/pixmaps/${pname}.png
+    substituteInPlace $out/share/applications/${pname}.desktop \
+      --replace 'Exec=AppRun --no-sandbox %U' 'Exec=${pname}' \
+      --replace 'Icon=launcher' 'Icon=${pname}'
```
Comment (SuperSandro2000) [inline pkgs/games/lunar-client/default.nix:27]: We shouldn't be using pname here so often

Comment (surfaceflinger) [reply pkgs/games/lunar-client/default.nix:27]: Should I just write lunar-client here explicitly?

Comment (SuperSandro2000) [reply pkgs/games/lunar-client/default.nix:27]: yeah

Comment (surfaceflinger) [reply pkgs/games/lunar-client/default.nix]: done!

**Review by SuperSandro2000 — APPROVED** [review]

**Review by SuperSandro2000 — APPROVED** [review]

**Pull request merged**